src/flowcean/environments/csv.py

```diff
@@ -41,3 +42,7 @@ def get_data(self) -> pl.DataFrame:
         if self.data is None:
             raise NotLoadedError
         return self.data
+
+    @staticmethod
+    def loader() -> LoaderFunction:
+        return lambda p: CsvDataLoader(p).load()
```

Comment:
Why do you need this wrapping function? The environment `__init__` methods already suffice the call signature of the `load_directory` function. Couldn't you just pass `CsvEnvironment` as an argument to `load_function`? Am I missing something?

Reply:
While the different offline envrionments share a similar `__init__` it is nowhere explicitly defined / inherited from the base class. I could ofcourse just *assume* that i can pass a path into an environment and then call `load` on it, but that's setup for failure if an environment has a different `__init__` method...
That's the reason for the `loader` function: To have a clear interface to create an environment from a path (if it's supported)...
With that I'm not sure if you can pass a (typed) constructor to another method in python...

Reply:
You can.

The interface is already defined by the `load_directory` function. This expects a function that takes a single argument (a path) to load an environment. If the constructor "accidentally" fits this signature, why introduce another wrapping `load` function?! Even without changing the interface of the `load_directory` function, the constructor already suffices that interface.

Reply:
I didn't know that you can just pass classes/the constructors around like this in python... It's tbh the first language I use that allows such things.
In that case the loader function is indeed superfluous an can be removed.
Shall we call `Load` from within the `load_directory` method? Otherwise the user has to write a loop just to call `load` on all environments.

Reply:
Is it necessary to load them at this stage? I think the helper function should not yet load the data.

Reply:
By definition we need to `load` the data before using it (e.g. passing it through a `TestTrainSplit`). In all our example codes we create the environment and directly afterwards call `load`. I would personally argue that `load` is not needed at all as part of the public interface and should be called once by `get_data`, but that's out-of-scope for this PR.
Right now you would need to loop through all environments and call `load` on them manually...

Reply:
But usually the user would call some variant of a stack or combine transform anyway?! And then it is a single environment with a single load function... I currently really like that we split initialization from loading data. This extremely speeds up early preprocessing of data and allows to check for structural bugs early on before even loading any data.